phoenix-core/src/test/java/org/apache/phoenix/index/IndexToolTest.java

```diff
@@ -0,0 +1,159 @@
+        String [] args =
+                IndexToolIT.getArgValues(true, true, schema,
+                        dataTable, indexTable, tenantId, IndexTool.IndexVerifyType.NONE,
+                        startTime , endTime);
+        CommandLine cmdLine = it.parseOptions(args);
+        it.populateIndexToolAttributes(cmdLine);
+        Assert.assertEquals(startTime, it.getStartTime());
+        Assert.assertEquals(endTime, it.getEndTime());
+    }
+
+    @Test
+    public void testParseOptions_timeRange_startTimeNotNull() {
+        Long startTime = 10L;
+        String [] args =
+                IndexToolIT.getArgValues(true, true, schema,
+                        dataTable, indexTable, tenantId, IndexTool.IndexVerifyType.NONE,
+                        startTime , null);
+        CommandLine cmdLine = it.parseOptions(args);
+        it.populateIndexToolAttributes(cmdLine);
+        Assert.assertEquals(startTime, it.getStartTime());
+        Assert.assertEquals(null, it.getEndTime());
+    }
+
+    @Test
+    public void testParseOptions_timeRange_endTimeNotNull() {
+        Long endTime = 15L;
+        String [] args =
+                IndexToolIT.getArgValues(true, true, schema,
+                        dataTable, indexTable, tenantId, IndexTool.IndexVerifyType.NONE,
+                        null , endTime);
+        CommandLine cmdLine = it.parseOptions(args);
+        it.populateIndexToolAttributes(cmdLine);
+        Assert.assertEquals(null, it.getStartTime());
+        Assert.assertEquals(endTime, it.getEndTime());
+    }
+
+    @Test
+    public void testParseOptions_timeRange_null() {
+        String [] args =
+                IndexToolIT.getArgValues(true, true, schema,
+                        dataTable, indexTable, tenantId, IndexTool.IndexVerifyType.NONE);
+        CommandLine cmdLine = it.parseOptions(args);
+        it.populateIndexToolAttributes(cmdLine);
+        Assert.assertNull(it.getStartTime());
+        Assert.assertNull(it.getEndTime());
+    }
+
+    @Test(timeout = 10000 /* 10 secs */)
+    public void testParseOptions_timeRange_startTimeInFuture() {
+        Long startTime = EnvironmentEdgeManager.currentTimeMillis() + 100000;
+        Long endTime = EnvironmentEdgeManager.currentTimeMillis() + 200000;
+        String [] args =
+                IndexToolIT.getArgValues(true, true, schema,
+                        dataTable, indexTable, tenantId, IndexTool.IndexVerifyType.NONE,
+                        startTime , endTime);
+        CommandLine cmdLine = it.parseOptions(args);
+        exceptionRule.expect(RuntimeException.class);
+        exceptionRule.expectMessage(INVALID_TIME_RANGE_EXCEPTION_MESSAGE);
+        it.populateIndexToolAttributes(cmdLine);
+    }
+
+    @Test(timeout = 10000 /* 10 secs */)
+    public void testParseOptions_timeRange_endTimeInFuture() {
+        Long startTime = EnvironmentEdgeManager.currentTimeMillis();
+        Long endTime = EnvironmentEdgeManager.currentTimeMillis() + 100000;
```

Comment:
This test didn't catch the bug in the time verification code since startTime is non-null here. TDD approach would be make startTime null here and observe the test failing then fix the verification code :)